corehq/apps/saved_reports/tests/test_scheduled_reports.py

```diff
@@ -170,23 +176,35 @@ def testMonthlyReportOnTheEndOfTheMonthDaysAfter31DontCount(self):
 class ScheduledReportSendingTest(TestCase):
 
     domain = 'test-scheduled-reports'
+    REPORT_NAME_LOOKUP = {
+        'worker_activity': 'corehq.apps.reports.standard.monitoring.WorkerActivityReport'
+    }
 
     @classmethod
     def setUpClass(cls):
         super(ScheduledReportSendingTest, cls).setUpClass()
         cls.domain_obj = create_domain(cls.domain)
+        cls.reports_role = UserRole.create(cls.domain, 'Test Role', permissions=Permissions(
+            view_report_list=[cls.REPORT_NAME_LOOKUP['worker_activity']]
+        ))
         cls.user = WebUser.create(
             domain=cls.domain,
             username='dummy@example.com',
             password='secret',
             created_by=None,
             created_via=None,
+            role_id=cls.reports_role._id
         )
 
+        with trap_extra_setup(ConnectionError, msg="cannot connect to elasicsearch"):
+            es = get_es_new()
+            initialize_index_and_mapping(es, CASE_INDEX_INFO)
```

Comment:
This will raise `SkipTest` if it cannot connect to ES, and I think `tearDownClass` will not be called in that scenario. Would it make sense to move it to the top of this method, possibly even before the `super` call?

Reply:
Hmm, thanks for digging into `trap_extra_setup`'s internals, I didn't realize it halted execution by raising an exception. I think it does make sense to hoist it above the `super()` call, but something about the approach still doesn't feel great -- I'll implement the change and see if I can think of anything cleaner

Reply:
This is a great point. I've now created a separate helper function (which could be a decorator) to verify that we can connect to ES before the test really starts. With the way `trap_extra_setup` was working, I did not like the idea of running `initialize_index_and_mapping` prior to calling `super()`.  It's much more granular now, but we do waste a ping to do the check.